python/hsfs/feature_group.py

```diff
@@ -1078,6 +1094,281 @@ def validate(
+            )
+
+        return self._feature_monitoring_result_engine.get_feature_monitoring_results(
+            config_name=config_name,
+            config_id=config_id,
+            start_time=start_time,
+            end_time=end_time,
+            with_statistics=with_statistics,
+        )
+
+    def _enable_scheduled_statistics_monitoring_fluent(
+        self,
+        name: str,
+        job_frequency: str = "DAILY",
+        feature_name: Optional[str] = None,
+        description: Optional[str] = None,
+        start_date_time: Optional[Union[int, str, datetime, date, pd.Timestamp]] = None,
+    ) -> "fmc.FeatureMonitoringConfig":
+        """Run a job to compute statistics on snapshot of feature data on a schedule.
+
+        !!! experimental
+            Public API is subject to change, this feature is not suitable for production use-cases.
+
+        !!! example
+            ```python3
+            # fetch feature group
+            fg = fs.get_feature_group(name="my_feature_group", version=1)
+
+            # enable scheduled statistics monitoring
+            my_config = fg._enable_scheduled_statistics_monitoring(
+                name="my_config",
+                job_frequency="DAILY",
+                start_date_time="2021-01-01 00:00:00",
+                description="my description",
+            ).with_detection_window(
+                # Statistics computed on 10% of the last week of data
+                time_offset="1w",
+                row_percentage=0.1,
+            ).save()
+            ```
+
+        # Arguments
+            name: Name of the feature monitoring configuration.
+                name must be unique for all configurations attached to the feature group.
+            feature_name: Name of the feature to monitor. If not specified, statistics
+                will be computed for all features.
+            job_frequency: Frequency at which to compute the statistics for the feature.
+                Options are "HOURLY", "DAILY", "WEEKLY", "MONTHLY", defaults to "DAILY".
+            description: Description of the feature monitoring configuration.
+            start_date_time: Start date and time from which to start computing statistics.
+
+        # Raises
+            `hsfs.client.exceptions.FeatureStoreException`.
+
+        # Return
+            `FeatureMonitoringConfig` Configuration with minimal information about the feature monitoring.
+                Additional information are required before feature monitoring is enabled.
+        """
+        if not self._id:
+            raise FeatureStoreException(
+                "Only Feature Group registered with Hopsworks can enable scheduled statistics monitoring."
+            )
+        fm_engine = feature_monitoring_config_engine.FeatureMonitoringConfigEngine(
+            feature_store_id=self._feature_store_id,
+            feature_group_id=self._id,
+        )
+
+        return fm_engine._build_default_scheduled_statistics_config(
```

Comment:
The config did not send to the backend.

Reply:
For now this is intended. Because we want to restrict the possible updates to the config to allow the history to be meaningful, we cannot just register it directly and then update it with a modified detection window. So they get a config back and need to call .save() to register it once they are done setting it up. Needs to be made clearer in docstring maybe

Reply:
yea just looking at the method name and the description, users do not know that they need to call `.save()` to commit the config.